### PR TITLE
[Kernel][Scheduler][MXNet] Scheduler for DGL kernels and MXNet backend support

### PR DESCRIPTION
**Pull request by ylfdq1118**
## Description
Update scheduler logic to use DGL kernels and support operators for MXNet backend. Also enable GPU CI.

## Checklist
- [x] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

## Changes
- [x] mxnet kernels
- [x] fix unit tests
- [x] fix tons of MSVC compiling issues...
- [x] scheduling logic
- [x] GPU CI

## TODO
- [ ] fix node flow
- [ ] enable all builtin function combinations and unit tests
- [ ] softmax module (edge_div_dst)

### CODE REVIEW

File: src/kernel/cpu/binary_bcast_reduce_sum.cc [hunk 1]
```diff
@@ -12,9 +12,9 @@ namespace kernel {
 #define REDUCER ReduceSum
 #define XPU kDLCPU
 
-EVAL(GEN_NDIM, GEN_DTYPE, GEN_TARGET, GEN_BINARY_OP, GEN_BCAST_DEFINE);
+EVAL(GEN_NDIM, GEN_DTYPE, GEN_TARGET, GEN_BINARY_OP, GEN_BCAST_DEFINE)
 EVAL(GEN_BACKWARD_MODE, GEN_NDIM, GEN_DTYPE, GEN_TARGET, GEN_BINARY_OP,
-     GEN_BACKWARD_BCAST_DEFINE);
+     GEN_BACKWARD_BCAST_DEFINE)
```
Comment (jermainewang) [inline src/kernel/cpu/binary_bcast_reduce_sum.cc]: Could you revert these kinds of changes? I'm afraid there will be too many conflicts when merging.

Comment (ylfdq1118) [reply src/kernel/cpu/binary_bcast_reduce_sum.cc]: Sure. I will checkout the old version of these files.

Comment (ylfdq1118) [reply src/kernel/cpu/binary_bcast_reduce_sum.cc]: Done

**Comment by ylfdq1118** [comment]
Close and change base

**Comment by ylfdq1118** [comment]
@jermainewang It's almost done. Check the TODO in description. Since I merged your softmax branch into this PR, there are probably too much change. I opened another PR in your fork (https://github.com/jermainewang/dgl/pull/1) and might be easier for you to review there.

But don't close this PR since I need it to trigger CI.

**Pull request merged**